packages/typespec-ts/test/integration/routes.spec.ts

```diff
@@ -32,13 +32,94 @@ describe("RoutesClient Rest Client", () => {
     assert.strictEqual(result.status, "204");
   });
 
-  it.skip("should have PathParameters SimpleExpansion Standard primitive", async () => {
+  it("should have allowReserved: true", async () => {
     const result = await client
-      .path("/routes/path/simple/standard/primitive{param}", "a")
+      .path("/routes/path/reserved-expansion/template/{param}", {
+        value: "foo/bar baz",
+        allowReserved: true
+      })
       .get();
     assert.strictEqual(result.status, "204");
   });
 
+  it("should have explode: true array", async () => {
+    const result = await client
+      .path("/routes/query/query-expansion/explode/array")
+      .get({
+        queryParameters: {
+          param: {
+            value: ["a", "b"],
+            explode: true,
+            style: "form"
+          }
+        }
+      });
+    assert.strictEqual(result.status, "204");
+  });
+
+  it("should have explode: true record", async () => {
```

Comment:
@timovv this case failed with 400 because we didn't prepare the object correctly.

![image](https://github.com/user-attachments/assets/55853589-2384-4dc0-8d95-06009c707a5a)
![image](https://github.com/user-attachments/assets/3810a633-da3a-42eb-b6d1-bf7412623221)